### DISCUSSION TITLE
Incomplete, Complete ... plus (optional) Final

### DISCUSSION THREAD

**Discussion by sanderjo** [sabnzbd/sabnzbd#2328] [Ideas]
Triggered by among others: https://forums.sabnzbd.org/viewtopic.php?p=128328#p128328

So, problem: Complete is on a NAS, and is not-yet-reachable, etc. That will cause problems for SAB & user.

I propose: an optional directory Final. If it is defined, SAB will move the downloaded & completed stuff from Complete to Final.
A post-processing script will run on the relevant directory: Complete (if Final not filled out), Final (if filled out).

AFAIK: most disk access is done to Incomplete, but Complete also has quite some disk access? With introducing Final, we avoid that that is done to a NAS or USB disk.

A non-working alternative would be a post-processing script "move_to_final.sh", but that would conflict with an existing pp-script

Normal-setups (disk local in system, and fast): user will not fill out Final
Where a NAS or slow final disk is involved, a user could fill out Final.

So:
Incomplete: C:\blabla\incomplete
Complete: C:\blabla\complete
Final: Z:\mystuff

**Comment by JackDandy** [comment]
> A post-processing script will run on the relevant directory: Complete (if Final not filled out), Final (if filled out).

Apologies if stating the obvious. to avoid an impact on existing client installations, can the value sent to **_post-processing scripts_** as "Completed" reflect whatever SAB determines as the "complete" or "final" destination path?

Also, considering network latency copying between SAB and "final" location, it could be useful if there was a `related_filename.lock` placed at the destination of where the file(s) will be copied, before the file starts to be transmitted, and then removed after the filecopy is actually done.
